pkg/sdn/plugin/networkpolicy.go

```diff
@@ -206,14 +199,39 @@ func (np *networkPolicyPlugin) syncNamespace(npns *npNamespace) {
 	otx := np.node.oc.NewTransaction()
 	otx.DeleteFlows("table=80, reg1=%d", npns.vnid)
 	if inUse {
-		if npns.isolated {
+		allPodsSelected := false
+
+		// Add "allow" rules for all traffic allowed by a NetworkPolicy
+		for _, npp := range npns.policies {
+			for _, flow := range npp.flows {
+				otx.AddFlow("table=80, priority=150, reg1=%d, %s actions=output:NXM_NX_REG2[]", npns.vnid, flow)
+			}
+			if npp.selectedIPs == nil {
+				allPodsSelected = true
+			}
+		}
+
+		if allPodsSelected {
+			// Some policy selects all pods, so all pods are "isolated" and no
+			// traffic is allowed beyond what we explicitly allowed above. (And
+			// the "priority=0, actions=drop" rule will filter out all remaining
+			// traffic in this Namespace).
+		} else {
+			// No policy selects all pods, so we need an "else accept" rule to
+			// allow traffic to pod IPs that aren't selected by a policy. But
+			// before that we need rules to drop any remaining traffic for any pod
+			// IP that *is* selected by a policy.
+			selectedIPs := sets.NewString()
 			for _, npp := range npns.policies {
-				for _, flow := range npp.flows {
-					otx.AddFlow("table=80, priority=100, reg1=%d, %s actions=output:NXM_NX_REG2[]", npns.vnid, flow)
+				for _, ip := range npp.selectedIPs {
+					if !selectedIPs.Has(ip) {
+						selectedIPs.Insert(ip)
+						otx.AddFlow("table=80, priority=100, reg1=%d, ip, nw_dst=%s, actions=drop", ip)
```

Comment:
The AddFlow()'s format string has one more format than the # of args given.  Not sure what the intention here is, to keep the VNID or remove it.